mods/screwdriver/init.lua

```diff
@@ -60,6 +60,23 @@ end
 
 screwdriver.rotate.colorfacedir = screwdriver.rotate.facedir
 
+local fourdir_tbl = {
+	[0] = 1, [1] = 2, [2] = 3, [3] = 0,
+}
+
+screwdriver.rotate["4dir"] = function(pos, node, mode)
+	if mode ~= screwdriver.ROTATE_FACE then
+		-- Can only rotate 4dir nodes in face mode
+		return nil
+	end
+	local rotation = node.param2 % 4 -- get first 2 bits
+	local other = node.param2 - rotation
+	rotation = fourdir_tbl[rotation] or 0
```

Comment:
The `or 0` here should be useless since `node.param2` always is an integer `>= 0` and due to the `% 4` `rotation` is one of `0`, `1`, `2`, `3` - `fourdir_tbl[rotation]` should thus never be `nil`.

You can get rid of the table entirely if you use `rotation = (rotation + 1) % 4`. I would also consider this to be more readable (add one more "step" to the rotation, wrap around at 4 steps because well, this is a rotation, and 4 is the equivalent of 360° here).